flink-connectors/flink-connector-aws-kinesis-firehose/src/main/java/org/apache/flink/connector/firehose/sink/KinesisFirehoseSinkWriter.java

```diff
@@ -65,6 +70,8 @@ private static SdkAsyncHttpClient createHttpClient(Properties firehoseClientProp
 
     private static FirehoseAsyncClient createFirehoseClient(
             Properties firehoseClientProperties, SdkAsyncHttpClient httpClient) {
+        AWSGeneralUtil.validateAwsConfiguration(firehoseClientProperties);
+        AWSGeneralUtil.validateWebIdentityTokenFileCredentialsProvider(firehoseClientProperties);
```

Comment:
Why are we calling an explicit validation for WebIdentifyToken here? Why not a generic top level one? This does not scale if more and more identify providers are added. Firehose should not care explicitly about which one to validate.